pkg/oci/dockerkeychain/docker_keychain.go

```diff
@@ -20,6 +21,54 @@ type DockerKeychain struct {
 	mutex        sync.Mutex
 }
 
+func NewDockerKeychains(ctx context.Context, apiReader client.Reader, namespaceName string, pullSecretsNames []string) (authn.Keychain, error) {
+	keychain := &DockerKeychain{}
+	err := keychain.loadDockerConfigFromSecrets(ctx, apiReader, namespaceName, pullSecretsNames)
+
+	return keychain, err
+}
+
+func (keychain *DockerKeychain) loadDockerConfigFromSecrets(ctx context.Context, apiReader client.Reader, namespaceName string, pullSecretsNames []string) error {
+	if len(pullSecretsNames) == 0 {
+		return nil
+	}
+
+	configFile := configfile.ConfigFile{
+		AuthConfigs: make(map[string]dockertypes.AuthConfig),
+	}
+
+	for _, pullSecretName := range pullSecretsNames {
+		pullSecret := corev1.Secret{}
+
+		if err := apiReader.Get(ctx, client.ObjectKey{Namespace: namespaceName, Name: pullSecretName}, &pullSecret); err != nil {
+			log.Info("registry pull secret not loaded", "name", pullSecretName, "namespace", namespaceName)
+
+			continue
+		}
+
+		dockerAuths, err := extractDockerAuthsFromSecret(&pullSecret)
+		if err != nil {
+			log.Info("failed to parse pull secret content", "name", pullSecret.Name, "namespace", pullSecret.Namespace)
+
+			return err
+		}
+
+		err = configFile.LoadFromReader(bytes.NewReader(dockerAuths))
```

Comment:
Food for thought: If multiple pull secrets contain credentials for the same registry, the last one wins, right?  How is the order determined and can we run into unexpected behaviour?

Reply:
I would call this user error to be honest, the other solution would be to compare them one by one by unpacking all of the pull secrets

Reply:
`PullSecretNames` function returns list of secrets in the following order: [dynakube-pull-secret , customPullSecret]. 
dynakube-pull-secret docker config is put into the map and then the one from customPullSecret overwriting the previous value. So customPullSecret has higher priority. 
IMHO customPullSecret for the tenant registry is unexpected scenario. In such a case download will fail or not if docker config is copied from dynakube-pull-secret.